### PR TITLE
Add Text Field Implementation

### PR DESCRIPTION
**Pull request by jcaracciolo**
Add TextField component with a given Implementation

### CODE REVIEW

**Comment by cristianames92** [comment]
Both `TextField` and `TextFieldRenderer` are in the main location and not under the correct one. Please reallocate those files.

File: Sources/Component.swift [hunk 1]
```diff
@@ -89,6 +101,9 @@ extension Component {
         case .label(let properties, let style, let layout):
             return .label(properties, style, layout)
             
+        case .textField(let properties, let style, let layout):
+            return .textField(properties.map(transform),style,layout)
```
Comment (guidomb) [inline Sources/Component.swift]: Add spaces between `,` and parameter. This should be changed to

```swift
return .textField(properties.map(transform), style, layout)
```

File: Sources/Component.swift [hunk 2]
```diff
@@ -21,6 +21,14 @@ public enum Gesture<MessageType> {
     
 }
 
+public enum TextFieldMessage {
```
Comment (guidomb) [inline Sources/Component.swift]: Delete this.

File: TextField.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+//  Created by Juan Franco Caracciolo on 4/10/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import Foundation
+
+public struct TextFieldProperties<MessageType> {
+    
+    public var text: String
+    public var textAfterLayout: String?
```
Comment (guidomb) [inline TextField.swift]: This should not be necessary for this component.

File: TextField.swift [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+//  PortalView
+//
+//  Created by Juan Franco Caracciolo on 4/10/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import Foundation
+
+public struct TextFieldProperties<MessageType> {
+
```
Comment (guidomb) [inline TextField.swift]: Missing `placeholder: String?` property.

File: TextField.swift [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        return TextFieldProperties<NewMessageType>(
+            text: self.text,
+            textAfterLayout: self.textAfterLayout,
+            onEvents: self.onEvents.map(transform)
+        )
+    }
+    
+}
+
+public struct OnTextFieldEvents<MessageType> {
```
Comment (guidomb) [inline TextField.swift]: This type should be an inner type inside `TextFieldProperties`

```swift
public struct TextFieldProperties<MessageType> {

  public struct Message {

    public let onBeginEditing: MessageType?
    public let onInputChange: MessageType?
    public let onEndEditing: MessageType?

  }

}
```

File: TextFieldRenderer.swift [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        let mailbox: Mailbox<MessageType> = textField.bindMessageDispatcher { mailbox in
+            properties.onEvents.onEditingBegin |> { _ = textField.dispatch(message: $0, for: .editingDidBegin, with: mailbox) }
+            properties.onEvents.onEditingChanged |> { _ = textField.dispatch(message: $0, for: .editingChanged, with: mailbox) }
+            properties.onEvents.onEditingEnd |> { _ = textField.dispatch(message: $0, for: .editingDidEnd, with: mailbox) }
+        }
+        
+        
+        return Render(view: textField, mailbox: mailbox) {
+            if let textAfterLayout = self.properties.textAfterLayout, let size = textField.maximumFontSizeForWidth() {
```
Comment (guidomb) [inline TextFieldRenderer.swift]: This is not necessary

**Review by guidomb — CHANGES_REQUESTED** [review]

File: TextField.swift (hunk 5 above)
Comment (jcaracciolo) [reply TextField.swift]: A new type cannot be nested inside a generic type like TextFieldProperties. [Reference](https://forums.developer.apple.com/thread/12298)

Comment (guidomb) [reply TextField.swift]: Yes it can. I've done it before. USE Xcode 8.3.1

Comment (jcaracciolo) [reply TextField.swift]: I will upgrade now and try it

**Review by guidomb — APPROVED** [review]

**Pull request merged**